compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -330,6 +331,13 @@ fn fill_place_recursively<'tcx>(
                 debug!(?i, ?inner_valtree);
 
                 let mut place_inner = match *ty.kind() {
+                    ty::Adt(def, substs) if !def.is_enum() => {
+                        let field = &def.variant(VariantIdx::from_usize(0)).fields[i];
+                        let field_ty = field.ty(tcx, substs);
+                        let projection_elem = ProjectionElem::Field(Field::from_usize(i), field_ty);
+
+                        ecx.mplace_projection(&place_adjusted, projection_elem).unwrap()
```

Comment:
This should be equivalent to mplace_field. I'm pretty sure all we do is call it inside `mplace_projection`